src/libcore/panic.rs

```diff
@@ -173,8 +173,14 @@ impl fmt::Display for PanicInfo<'_> {
 ///
 /// panic!("Normal panic");
 /// ```
+///
+/// # Comparisons
+///
+/// Comparisons for equality and ordering are made in file, line, then column priority. Such
+/// comparisons can occasionally have surprising results. See [`Location::file`]'s documentation for
+/// more discussion.
```

Comment:
```suggestion
/// Comparisons for equality and ordering are made in file, line, then column priority.
/// Files are compared as strings, not `Path`, which could be unexpected.
/// See [`Location::file`]'s documentation for more discussion.
```

Let’s mention string ordering explicitly here?